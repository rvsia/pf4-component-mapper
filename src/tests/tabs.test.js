import React from 'react';
import toJson from 'enzyme-to-json';
import Tabs from '../form-fields/tabs';
import { mount, shallow } from 'enzyme';

describe('Tabs component', () => {
  const props = {
    fields: [
      {
        key: 'cosiKey',
        title: 'cosiTitle',
        name: 'cosiName',
        fields: [],
      },
      {
        key: 'cosiKey2',
        title: 'cosiTitle2',
        name: 'cosiName2',
        fields: [],
      },
    ],
    formOptions: {
      renderForm: (fields, formOptions) => <div>{ 'Here would be form' }</div>,
    },
  };

  it('should render tabs correctly', () => {
    const wrapper = shallow(
      <Tabs { ...props }>
      </Tabs>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should switch tabs correctly', () => {
    const wrapper = mount(<Tabs { ...props } />);
    expect(wrapper.instance().state.activeTabKey).toEqual(0);

    const secondTabButton = wrapper.find('[className="pf-c-tabs__button"]').at(1);
    secondTabButton.simulate('click');

    expect(wrapper.instance().state.activeTabKey).toEqual(1);
  });
});
