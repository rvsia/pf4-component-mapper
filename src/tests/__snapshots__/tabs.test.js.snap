// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component should render tabs correctly 1`] = `
<Fragment
  key="cosiKey"
>
  <Title
    className=""
    headingLevel="h1"
    size="3xl"
  >
    cosiTitle
  </Title>
  <div>
    Here would be form
  </div>
</Fragment>
`;
